core/qt-ble.cpp

```diff
@@ -256,20 +318,24 @@ dc_status_t qt_ble_open(dc_custom_io_t *io, dc_context_t *context, const char *d
 
 	if (!list.isEmpty()) {
 		const QLowEnergyCharacteristic &c = list.constLast();
-		QList<QLowEnergyDescriptor> l = c.descriptors();
 
-		qDebug() << "Descriptor list with" << l.length() << "elements";
+		if (device_is_hw(io->user_device)) {
+			ble->setupHwTerminalIo(list);
+		} else {
+			QList<QLowEnergyDescriptor> l = c.descriptors();
 
-		QLowEnergyDescriptor d;
-		foreach(d, l)
-			qDebug() << "Descriptor:" << d.name() << "uuid:" << d.uuid().toString();
+			qDebug() << "Descriptor list with" << l.length() << "elements";
 
+			QLowEnergyDescriptor d;
+			foreach(d, l)
+				qDebug() << "Descriptor:" << d.name() << "uuid:" << d.uuid().toString();
 
-		if (!l.isEmpty()) {
-			d = l.first();
-			qDebug() << "now writing \"0x0100\" to the first descriptor";
+			if (!l.isEmpty()) {
+				d = l.first();
+				qDebug() << "now writing \"0x0100\" to the first descriptor";
```

Comment:
This last whitespace look wrong, but is is not ... there is an if clause added around, so 1 tab more indention.